internal/dag/dag.go

```diff
@@ -395,6 +399,24 @@ func processIngressRoute(ir *ingressroutev1.IngressRoute, prefixMatch string, vi
 	}
 }
 
+// matchesPathPrefix checks whether the given path matches the given prefix
+func matchesPathPrefix(path, prefix string) bool {
+	if len(prefix) == 0 {
+		return true
+	}
+	// an empty string cannot have a prefix
+	if len(path) == 0 {
+		return false
+	}
+	if prefix[len(prefix)-1] != '/' {
+		prefix = prefix + "/"
+	}
+	if path[len(path)-1] != '/' {
+		path = path + "/"
+	}
+	return strings.HasPrefix(strings.ToLower(path), strings.ToLower(prefix))
```

Comment:
Please revert the ToLower, that's not part of #488. If we want matching to be case insensitive then that should be a feature request (spoiler, urls are case sensitive, I don't think we should do this).

Reply:
Ah! Hostnames are case-insensitive, but URLs are case-sensitive. Thanks @davecheney